### PR TITLE
Upgrade glob

### PR DESCRIPTION
**Pull request by yufeih**
The latest version of glob contains bug fixes we need

### CODE REVIEW

**Comment by yufeih** [comment]
Pending on https://github.com/kthompson/glob/issues/39

File: test/docfx.Test/lib/GlobUtilityTest.cs [hunk 1]
```diff
@@ -69,10 +72,6 @@ public void MatchFilesUsingGlobPattern(string pattern, string files, bool match)
         [InlineData("a[\\\\b]c")]
         [InlineData("{{a,b}}")]
         [InlineData("z{a,b{,c}d")]
-        [InlineData("**/")]
-        [InlineData("\\[*")]
-        [InlineData("a{b,c,d}e{d}{}")]
```
Comment (OsmondJiang) [inline test/docfx.Test/lib/GlobUtilityTest.cs]: we are not going to support these?

Comment (yufeih) [reply test/docfx.Test/lib/GlobUtilityTest.cs]: these are supported

Comment (OsmondJiang) [reply test/docfx.Test/lib/GlobUtilityTest.cs]: would we add these supported test cases to above test?

---
In reply to: [229163945](https://github.com/dotnet/docfx/pull/3541#discussion_r229163945) [](ancestors = 229163945)

Comment (OsmondJiang) [reply test/docfx.Test/lib/GlobUtilityTest.cs]: liken this one: a{b,c,d}e{d}{}

---
In reply to: [229583124](https://github.com/dotnet/docfx/pull/3541#discussion_r229583124) [](ancestors = 229583124,229163945)

Comment (yufeih) [reply test/docfx.Test/lib/GlobUtilityTest.cs]: sure

**Pull request merged**